airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -87,6 +91,19 @@ def get_error_display_message(self, exception: BaseException) -> Optional[str]:
         """
         return None
 
+    def read_records_as_messages(
```

Comment:
It's a bit smelly that I can override `read_records` and `read_records_as_messages`, but only the latter ever gets called

Are we just saying that a developer can return either `Mapping` or `AirbyteMessage` from `read_records`? Should we just change the type of that method to be a union, and move this logic into AbstractSource so that it wraps any `Mappings` into `AirbyteMessage`s?

Reply:
@brianjlai the `Stream` interface has to return the message because we'll want to return more types of `AirbyteMessage`s (like logs and tracemessages)

@sherifnada 
In Java-world, I would change the scope of `read_records` to protected so it can be overridden by subclasses, but it shouldn't be part of the interface (the source only calls `read_records_as_messages` now)

> Are we just saying that a developer can return either Mapping or AirbyteMessage from read_records?
No, `read_records` always returns a Mapping. The mapping gets converted to an [AirbyteRecord on line 105](https://github.com/airbytehq/airbyte/pull/18572/files#diff-49f3635a2319e7b734cc81dd557f7d14fa8b7501f981a46855bc9cee154607a0R105)